### PR TITLE
Integrate wasmvm v1.2.0

### PR DESCRIPTION
**Pull request by alpe**
See #1143

Uses ~~[wasmvm v1.2.0-rc.1](https://github.com/CosmWasm/wasmvm/releases/tag/v1.2.0-rc.1)~~ [wasmvm v1.2.0](https://github.com/CosmWasm/wasmvm/releases/tag/v1.2.0)

### CODE REVIEW

File: x/wasm/client/cli/gov_tx_test.go [hunk 1]
```diff
@@ -104,7 +104,7 @@ func TestParseCodeInfoFlags(t *testing.T) {
 	wasmBin, err := os.ReadFile("../../keeper/testdata/hackatom.wasm")
 	require.NoError(t, err)
 
-	checksumStr := "13a1fc994cc6d1c81b746ee0c0ff6f90043875e0bf1d9be6b7d779fc978dc2a5"
+	checksumStr := "3A8C25F7C4E6CFCA07C388FADCFBE4646C8FF1D04715D05ED8B25253DB3865A1"
```
[the remark this answers is not in the document]
Comment (alpe) [reply x/wasm/client/cli/gov_tx_test.go:107]: I copied the hash. It does not make a difference if upper or lower case.

File: x/wasm/keeper/handler_plugin_encoders_test.go [hunk 2]
```diff
@@ -559,24 +628,174 @@ func TestEncoding(t *testing.T) {
 			output: []sdk.Msg{
 				&govtypes.MsgVote{
 					ProposalId: 1,
-					Voter:      addr1.String(),
+					Voter:      myAddr.String(),
 					Option:     govtypes.OptionNoWithVeto,
 				},
 			},
 		},
+		"Gov vote: unset option": { // regression test, unset = yes
```
[the remark this answers is not in the document]
Comment (alpe) [reply x/wasm/keeper/handler_plugin_encoders_test.go:636]: This is a regression test to cover this edge case. It will fail if `yes` is not the first element (=`0`) anymore so that we get notified of a breaking change. This setup is not ideal. Instead the first element should be 'Undefined' or `Yes = iota +1` in Go so that this case can be identified and handled proper.

File: x/wasm/keeper/handler_plugin_encoders.go [hunk 3]
```diff
@@ -304,13 +344,10 @@ func EncodeGovMsg(sender sdk.AccAddress, msg *wasmvmtypes.GovMsg) ([]sdk.Msg, er
 		option = govtypes.OptionNoWithVeto
 	case wasmvmtypes.Abstain:
 		option = govtypes.OptionAbstain
+	default:
+		option = govtypes.OptionEmpty
```
[the remark this answers is not in the document]
Comment (alpe) [reply x/wasm/keeper/handler_plugin_encoders.go:348]: Empty is not a valid vote option in the [SDK](https://github.com/cosmos/cosmos-sdk/blob/v0.45.12/x/gov/types/vote.go#L121). This will result in an error. I can make this more explicit here.

File: x/wasm/keeper/handler_plugin_encoders_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (webmaster128) [reply x/wasm/keeper/handler_plugin_encoders_test.go:636]: Good, happy to change that in wasmvm

[the remark this answers is not in the document]
Comment (webmaster128) [reply x/wasm/keeper/handler_plugin_encoders_test.go:636]: Is this better? https://github.com/CosmWasm/wasmvm/pull/392

**Review by webmaster128 — APPROVED** [review]
💪
